engines/freescape/games/driller/dos.cpp

```diff
@@ -155,86 +155,39 @@ Graphics::ManagedSurface *DrillerEngine::load8bitTitleImage(Common::SeekableRead
 
 	int i = 0;
 	int j = 0;
-	int command = -1;
-	int singlePixelsToProcess = 0;
-	bool repeatedPixelsToProcess = false;
 	file->seek(offset);
 	while (!file->eos()) {
-		assert(i <= 320);
-		int pixels = -1;
-		int repetition = -1;
-
-		if (singlePixelsToProcess == 0 && !repeatedPixelsToProcess) {
-			if (command < 0)
-				command = file->readByte();
-
-			//debug("reading command: %x at %lx", command, file->pos() - 1);
-
-			assert(command >= 0x7f);
-			singlePixelsToProcess = (0xff - command + 2) * 2;
-			//debug("single Pixels to process: %d", singlePixelsToProcess);
-
-			repeatedPixelsToProcess = true;
-			if (i == 320) {
-				j++;
-				i = 0;
-			}
-			command = -1;
-			continue;
-		}
-
-		if (singlePixelsToProcess > 0) {
-			singlePixelsToProcess--;
-			pixels = file->readByte();
-			//debug("reading pixels: %x at %d, %d", pixels, i, j);
-			renderPixels8bitTitleImage(surface, i, j, pixels);
-		} else if (repeatedPixelsToProcess) {
-			repetition = file->readByte() + 1;
-			//debug("reading repetition: %x", repetition - 1);
-			assert(repetition > 0);
-			if (repetition >= 0x80) {
-				command = repetition - 1;
-				repeatedPixelsToProcess = false;
-				continue;
-			}
-
-			if (i == 320) {
-				j++;
-				i = 0;
-				continue;
+		int command = file->readByte();
+		if (command & 0x80) {
+			//Copy N bytes verbatim
+			int repeat = (0xFF - command + 2) * 2;
+			while (repeat--) {
+				int pixels = file->readByte();
+				renderPixels8bitTitleImage(surface, i, j, pixels);
+				if (i == 320) {
+					i = 0;
+					j++;
+					if (j == 200)
+						return surface;
+					int eol = file->readByte();
+					assert(eol == 2);
```

Comment:
Despite checking for the `0x02` byte at the end of the line, it is actually a "start-of-line" marker (and after line 200, there is no `0x02` marker, but an `0x00` end-of-file marker instead). So this check should actually be moved forward, and the offset into `CGATITLE.RL` and `EGATITLE.RL` should not be `0x1B3` but `0x1B2` instead.

Reply:
Feel free to move it, it will make more sense there.